vowpalwabbit/parser.cc

```diff
@@ -571,6 +575,14 @@ void enable_sources(vw& all, bool quiet, size_t passes, input_options& input_opt
         all.example_parser->reader = VW::parsers::flatbuffer::flatbuffer_to_examples;
       }
 #endif
+
+#ifdef BUILD_EXTERNAL_PARSER
+      else if (input_options.external_parser)
+      {
+        all.external_parser = VW::make_unique<VW::external_parser>(&all);
+        all.example_parser->reader = VW::parse_examples;
```

Comment:
Don't you need to call into the external parser somewhere?

Reply:
vw calls the parser you provide [here](https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/vowpalwabbit/parse_dispatch_loop.h#L22) during parsing

Reply:
the bindings would probably not work with this, bc this is being extended on compile time. You would need to compile another python package with this underlaying forked vw binary.

Reply:
Regarding the idea of more extendable args, it could be possible if you have a generic hook for 'plugins' where vw has a call_external_arg_parser() and as long as a plugin fulfills that contract, it could work. This way the new flag would be defined on the other side, and the end result should look the same. That would allow you to have more generic names on flags that you can later migrate to vw itself, without breaking whoever was consuming the forked binary before.

Reply:
added `VW::external::parser::set_parse_args()` in `parse_args.cc`

external parser adds whatever options it wants see [here](https://github.com/VowpalWabbit/reinforcement_learning/pull/265/files#diff-5617db68ef0406524f96b6a49b8526b756796e20e65a3fb31019947f924826faR23)

the contract that the external parser has to fill (in general, not only for args) can be seen [here](https://github.com/VowpalWabbit/reinforcement_learning/pull/265/files#diff-959fe33f0cd6431dc062b3357fda604b7b3fb4e47aad4fc1d25e31ddf3e42327R1)